File: app/src/main/java/com/jeronimo/swingradio/MainViewPagerAdapter.java
package com.jeronimo.swingradio;

import android.app.Fragment;
import android.app.FragmentManager;
import android.support.v13.app.FragmentPagerAdapter;

/**
 * Created by gleb on 23.12.15.
 */
public class MainViewPagerAdapter extends FragmentPagerAdapter {

    private PlayFragment playFragment;
    private BlogFragment blogFragment;
    private static int NUM_ITEMS = 2;
    public MainViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {

        switch (position) {
            case 0:
                if(playFragment == null)
                {
                    playFragment = new PlayFragment();
                }
                return playFragment;
            case 1:
                if(blogFragment == null)
                {
                    blogFragment = new BlogFragment();
                }
                return blogFragment;
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return NUM_ITEMS;
    }
}

File: app/src/main/java/com/jeronimo/swingradio/MainViewPagerAdapter.kt
package com.jeronimo.swingradio

import android.app.Fragment
import android.app.FragmentManager
import android.support.v13.app.FragmentPagerAdapter

/**
 * Created by gleb on 23.12.15.
 */
class MainViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    private var playFragment: PlayFragment? = null
    private var blogFragment: BlogFragment? = null

    override fun getItem(position: Int): Fragment? {

        when (position) {
            0 -> {
                if (playFragment == null) {
                    playFragment = PlayFragment()
                }
                return playFragment
            }
            1 -> {
                if (blogFragment == null) {
                    blogFragment = BlogFragment()
                }
                return blogFragment
            }
            else -> return null
        }
    }

    override fun getCount(): Int {
        return NUM_ITEMS
    }

    companion object {
        private val NUM_ITEMS = 2
    }
}
